### PR TITLE
Api

### PR DESCRIPTION
**Pull request by david-yz-liu**

### CODE REVIEW

File: api.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	);
+    // set the PDO error mode to exception
+    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+    $stmt = $conn->prepare($query);
+    $stmt->execute();
+
+	if($method == "GET"){
+		//fetch all results
+		$fetched = $stmt->fetchAll();
+		if (count($query) >= 1) {
```
Comment (david-yz-liu) [inline api.php]: You don't need this check. Just set `$result['DATA'] = $fetched`

File: error_check.php [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	$selected_table = NULL;
+	switch($table){
+		case 'course_question_choice':
+			if ($operation!='delete'){
+				$selected_table = $course_question_choice;
+			}
+			break;
+
+		case 'courses':
+			if ($operation!='put'){
```
Comment (david-yz-liu) [inline error_check.php]: Remove the `put` checks from this function.

File: error_check.php [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		if($i == count($url)){
+			$result['table'] = $key;
+		}
+	}
+	$result['condition'] = substr($result['condition'], 0,-4);
+	return $result;
+}
+
+/**
+ * Checks if column exists within database.
```
Comment (david-yz-liu) [inline error_check.php]: Say that you check whether the given table exists, the given columns are in that table, and (at least for now) whether the operation is allowed on that table.

File: error_check.php [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+		}
+	}
+
+	foreach($columns as $column){
+		if (!in_array($column, $selected_table)){
+			throw new Exception('Column: '. $column . ' does not exist');
+		}
+	}
+
+	if ($operation=='put'){
```
Comment (david-yz-liu) [inline error_check.php]: (So for now you can make this an issue, but don't include it in this version of the code.)

**Comment by richardzeng1** [comment]
#4  #18 #20 #22

**Comment by richardzeng1** [comment]
#25

**Pull request merged**